### PR TITLE
Simple notifications

### PR DESCRIPTION
**Pull request by ushkarev**
This is a bit of a complicated change :(

Because the plan for notifications radically changed during development, there is work here that's been undone. It seemed like a good idea to not drop the history of the older version of notifications for reference as there's a chance that some of it will come back in future iterations.

Everything before `Merge branch 'notifications-master' into simple-notifications` had already been reviewed and was awaiting merging, so it's only the subsequent changes that are of interest to this PR.

### CODE REVIEW

**Comment by ushkarev** [comment]
Depends on [api#521](https://github.com/ministryofjustice/money-to-prisoners-api/pull/521)

**Comment by ushkarev** [comment]
Main changes since the previous iteration of notifications:
- removal of time-based security profile totals
- removal of most notification rules (only monitored prisoners and payment sources remain)
- display of notifications is now grouped per-date rather than last xx

File: mtp_noms_ops/apps/security/tests/test_views.py [hunk 1]
```diff
@@ -1222,19 +1229,306 @@ def test_pin_profile(self):
...
+            self.login(follow=False, rsps=rsps)
+            response = self.client.get(reverse('security:notification_list'), follow=True)
+        self.assertNotContains(response, '<!-- security:notification_list -->')
+
+    def login_with_pilot_user(self, rsps):
+        self.login(user_data=self.get_user_data(flags=[
+            notifications_pilot_flag, hmpps_employee_flag, confirmed_prisons_flag,
+        ]), rsps=rsps)
+
+    def test_no_notifications_not_monitoring(self):
```
Comment (marcofucci) [inline mtp_noms_ops/apps/security/tests/test_views.py]: A bit of docstrings for these tests would help as it takes time to figure out that they are expecting.

File: mtp_noms_ops/apps/security/views/object_detail.py [hunk 2]
```diff
@@ -101,15 +100,7 @@ class SenderDetailView(SecurityDetailView):
...
-        try:
-            return detail_object['bank_transfer_details'][0]['sender_name']
-        except (KeyError, IndexError):
-            pass
-        try:
-            return detail_object['debit_card_details'][0]['cardholder_names'][0]
-        except (KeyError, IndexError):
-            pass
-        return _('Unknown sender')
+        return sender_profile_name(detail_object) or _('Unknown sender')
```
Comment (marcofucci) [inline mtp_noms_ops/apps/security/views/object_detail.py]: You have a similar line in `security.forms.object_list` so maybe you could reuse some logic?

File: mtp_noms_ops/apps/security/forms/object_list.py [hunk 3]
```diff
@@ -588,3 +590,120 @@ def describe_field_amount_pattern(self):
...
+                    details['disbursement_ids'].add(event['disbursement_id'])
+
+            if event['prisoner_profile']:
+                profile = event['prisoner_profile']
+                if profile['id'] in date_group['prisoners']:
+                    details = date_group['prisoners'][profile['id']]
+                else:
+                    details = self.make_date_group_profile(
+                        profile['id'],
+                        profile['prisoner_name'] + ' (' + profile['prisoner_number'] + ')'
```
[the remark this answers is not in the document]
Comment (ushkarev) [reply mtp_noms_ops/apps/security/forms/object_list.py]: but then we're mixing `'` and `"` ugh the difficult choices developers need to make :(

File: mtp_noms_ops/apps/security/tests/test_views.py [hunk 4]
```diff
@@ -1222,19 +1229,306 @@ def test_pin_profile(self):
...
+                        'last_result_count': 4,
+                        'site_url': '/en-gb/security/prisoners/1/',
+                        'filters': [{'field': 'ordering', 'value': '-received_at'}],
+                    },
+                )
+
+
+class NotificationsTestCase(SecurityBaseTestCase):
+    def test_cannot_access_without_pilot_flag(self):
+        with responses.RequestsMock() as rsps:
```
[the remark this answers is not in the document]
Comment (ushkarev) [reply mtp_noms_ops/apps/security/tests/test_views.py]: `@responses.activate` is not as good. the context manager asserts that the requests are made. i want to convert all the tests, but not in this pr

File: mtp_noms_ops/apps/security/forms/object_list.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcofucci) [reply mtp_noms_ops/apps/security/forms/object_list.py]: yeah I think it's accepted in these cases and flake8-commas shouldn't complain.

File: mtp_noms_ops/apps/security/forms/object_list.py [hunk 5]
```diff
@@ -588,3 +590,148 @@ def describe_field_amount_pattern(self):
...
+            del filters['offset']
+            del filters['limit']
+            if data['newest']:
+                filters['triggered_at__lt'] = parse_date(data['newest']) + datetime.timedelta(days=1)
+                filters['triggered_at__gte'] = parse_date(data['oldest'])
+        return filters
+
+    def get_object_list(self):
+        events = convert_date_fields(super().get_object_list())
+        date_groups = map(summarise_date_group, group_events_by_date(events))
```
Comment (marcofucci) [inline mtp_noms_ops/apps/security/forms/object_list.py]: Non-critical: I think nowadays python devs prefer list comprehension to map/filter as the former is clearer and more explicit.

**Pull request merged**